pgx/src/callbacks.rs

```diff
@@ -310,6 +311,7 @@ where
 {
     static mut SUB_HOOKS: Option<SubCallbackMap> = None;
 
+    #[pg_guard]
```

Comment:
As mentioned above this applies `#[no_mangle]` -- totally wrong for callbacks. This will collide with any function named `callback` in C, or any other `#[pg_guard]` function named `callback`, etc. I think we should not apply `#[no_mangle]` in `#[pg_guard]`, but we could also just not use it here.